lib/oli/delivery/gating.ex

```diff
@@ -0,0 +1,120 @@
+defmodule Oli.Delivery.Gating do
+  @moduledoc """
+  The Delivery.Gating context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Oli.Repo
+
+  alias Oli.Delivery.Gating.GatingCondition
+
+  @doc """
+  Returns the list of gating_conditions.
+
+  ## Examples
+
+      iex> list_gating_conditions()
+      [%GatingCondition{}, ...]
+
+  """
+  def list_gating_conditions do
+    Repo.all(GatingCondition)
+  end
+
+  @doc """
+  Returns the list of gating_conditions for the list of given resource ids
+
+  ## Examples
+
+      iex> list_gating_conditions([1,2,3])
+      [%GatingCondition{}, ...]
+
+  """
+  def list_gating_conditions(resource_ids) do
```

Comment:
This must also include a section or section_slug as an argument, otherwise it will return gatingconditions across all course sections that utilize a particular resource.